python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -162,8 +161,17 @@ def read_fn():
         self.last_update_time = 0.0
         self.update_interval_s = update_interval_s
 
-        # Tracks active worker nodes
+        # The following block of data on active nodes is filled once per
+        # autoscaler update in `self.update_node_lists()`.
+        # Managed worker nodes (node kind "worker"):
         self.workers = []
+        # Managed and unmanaged worker nodes
+        # (node kinds "worker" and "unmanaged"):
+        self.all_workers = []
+        # All nodes (workers, unmanaged workers, and head)
+        self.all_nodes = []
+        self.head_node_ip = ""
```

Comment:
Should this be None?

Reply:
I guess that's a more conventional for denoting a thing you haven't set yet.

Reply:
Replaced with a "head_node" item initialized to None.